include/triton/core/tritonserver.h

```diff
@@ -905,6 +905,34 @@ TRITONSERVER_InferenceRequestAppendInputData(
     const void* base, size_t byte_size, TRITONSERVER_MemoryType memory_type,
     int64_t memory_type_id);
 
+/// Assign a buffer of data to an input for execution on a particular device.
+/// The buffer will be appended to any existing buffers for that input on that
+/// device. The 'inference_request' object takes ownership of the buffer and so
+/// the caller should not modify or free the buffer until that ownership is
+/// released by 'inference_request' being deleted or by the input being removed
+/// from 'inference_request'.
+/// If the execution is scheduled on a device that does not have a input buffer
+/// specified using this function, then the input buffer specified with
+/// TRITONSERVER_InferenceRequestAppendInputData will be used. 
+///
+/// \param inference_request The request object. 
+/// \param name The name of the input.
+/// \param base The base address of the input data.
+/// \param byte_size The size, in bytes, of the input data.
+/// \param memory_type The memory type of the input data.
+/// \param memory_type_id The memory type id of the input data.
+/// \param device_kind All model instances executing on the device specified by 
+/// device_kind/device_id will use this input for execution.
+/// \param device_id All model instances executing on the device specified by 
+/// device_kind/device_id will use this input for execution. 
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC TRITONSERVER_Error*
+TRITONSERVER_InferenceRequestAppendInputDataForDevice(
```

Comment:
Will `TRITONSERVER_InferenceRequestRemoveAllInputData` clear the data appended by this API? I think it should

Reply:
Yes it will. I will update the docs. The implementation already does this in the PR to the server repo